airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams/streams.py

```diff
@@ -171,55 +189,65 @@ class AdAccount(FBMarketingStream):
     use_batch = False
     enable_deleted = False
 
-    def get_task_permissions(self) -> Set[str]:
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._fields_dict = {}
+
+    def get_task_permissions(self, account_id: str) -> Set[str]:
         """https://developers.facebook.com/docs/marketing-api/reference/ad-account/assigned_users/"""
         res = set()
         me = User(fbid="me", api=self._api.api)
         for business_user in me.get_business_users():
-            assigned_users = self._api.account.get_assigned_users(params={"business": business_user["business"].get_id()})
+            assigned_users = self._api.account(account_id=account_id).get_assigned_users(
+                params={"business": business_user["business"].get_id()}
+            )
             for assigned_user in assigned_users:
                 if business_user.get_id() == assigned_user.get_id():
                     res.update(set(assigned_user["tasks"]))
         return res
 
-    @cached_property
-    def fields(self) -> List[str]:
-        properties = super().fields
+    def fields(self, account_id: str, **kwargs) -> List[str]:
+        if self._fields_dict.get(account_id):
+            return self._fields_dict.get(account_id)
+
+        properties = super().fields(**kwargs)
         # https://developers.facebook.com/docs/marketing-apis/guides/javascript-ads-dialog-for-payments/
         # To access "funding_source_details", the user making the API call must have a MANAGE task permission for
         # that specific ad account.
-        permissions = self.get_task_permissions()
+        permissions = self.get_task_permissions(account_id=account_id)
         if "funding_source_details" in properties and "MANAGE" not in permissions:
             properties.remove("funding_source_details")
         if "is_prepay_account" in properties and "MANAGE" not in permissions:
             properties.remove("is_prepay_account")
+
+        self._fields_dict[account_id] = properties
         return properties
 
-    def list_objects(self, params: Mapping[str, Any]) -> Iterable:
+    def list_objects(self, params: Mapping[str, Any], account_id: str) -> Iterable:
         """noop in case of AdAccount"""
-        fields = self.fields
+        fields = self.fields(account_id=account_id)
         try:
-            return [FBAdAccount(self._api.account.get_id()).api_get(fields=fields)]
+            return [FBAdAccount(self._api.account(account_id=account_id).get_id()).api_get(fields=fields)]
```

Comment:
If we are passing in the account id do we need to get the account and then get the id back from it? Can we potentially just `FBAdAccount(account_id).api_get(fields=fields)`?

Reply:
`get_id()` returns id in format of `act_01234567890` instead of just `0123456789`. As this ID string is formed by the Facebook library I would leave it as it is.

Reply:
ah got it, i thought this was something our api side was handling. all good